srml/support/procedural/src/lib.rs

```diff
@@ -40,23 +40,47 @@ use proc_macro::TokenStream;
+/// with `Store` a (pub) trait generated associating each storage item to the `Module` and
+/// `as Example` setting the prefix used for storage items of this module. `Example` must be unique:
+/// another module with the same name and the same inner storage item name will conflict.
 ///
 /// Basic storage consists of a name and a type; supported types are:
 ///
 /// * Value: `Foo: type`: Implements [StorageValue](../srml_support/storage/trait.StorageValue.html).
-/// * Map: `Foo: map type => type`: implements [StorageMap](../srml_support/storage/trait.StorageMap.html)
-/// * Linked map: `Foo: linked_map type => type`: Implements [StorageMap](../srml_support/storage/trait.StorageMap.html)
-/// and [EnumarableStorageMap](../srml_support/storage/trait.EnumerableStorageMap.html).
-/// * Double map: `Foo: double_map u32, $hash(u32) => u32;`: Implements `StorageDoubleMap` with `$hash` representing a
-/// choice of hashing algorithms available in [`Hashable` trait](../srml_support/trait.Hashable.html).
+/// * Map: `Foo: map hasher($hash) type => type`: Implements [StorageMap](../srml_support/storage/trait.StorageMap.html)
+///   with `$hash` representing a choice of hashing algorithms available in the
+///   [`Hashable` trait](../srml_support/trait.Hashable.html).
 ///
-///   /!\ Be careful when choosing the hash function, malicious actors could craft second keys to lower the trie.
+///   `hasher($hash)` is optional and its default is blake2_256.
 ///
-/// And it can be extended as such:
+///   /!\ Be careful with each key in the map that is inserted in the trie `$hash(module_name ++ storage_name ++ key)`.
+///   For an untrusted key, a cryptographic hasher such as blake2_256 must be used.
+///   Otherwise, other values in storage can be compromised.
+///
+/// * Linked map: `Foo: linked_map hasher($hash) type => type`: Same as `Map` but also implements
+///   [EnumarableStorageMap](../srml_support/storage/trait.EnumerableStorageMap.html).
+///
+/// * Double map: `Foo: double_map hasher($hash) u32, $hash2(u32) => u32`: Implements `StorageDoubleMap` with
+///   `$hash` and `$hash2` representing choices of hashing algorithms available in the
+///   [`Hashable` trait](../srml_support/trait.Hashable.html).
+///
+///   `hasher($hash)` is optional and its default is blake2_256.
+///
+///   /!\ Be careful with each key pair in the double map that is inserted in the trie.
+///   The final key is calculated as follows:
+///
+///   ```nocompile
+///   $hash(module_name ++ storage_name ++ first_key) ++ $hash2(second_key)
+///   ```
+///
+///   If the first key is untrusted, a cryptographic hasher such as blake2_256 must be used.
+///   Otherwise, other values of all storage items can be compromised.
+///
+///   If the second key is untrusted, a cryptographic hasher such as blake2_256 must be used.
+///   Otherwise, malicious actors could craft second keys to lower the trie.
```

Comment:
Hmm I'm mixed this was the old message but shouldn't it be 
```suggestion
///   Otherwise, other values of same first key in the storage can be compromised.
```